src/containerapp/azext_containerapp/custom.py

```diff
@@ -4332,3 +4339,258 @@ def delete_workload_profile(cmd, resource_group_name, env_name, workload_profile
+                                targetContainerName=inspect_result["targetContainerName"],
+                                targetContainerAppName=inspect_result["targetContainerAppName"],
+                                targetContainerAppEnvironmentName=inspect_result["targetContainerAppEnvironmentName"],
+                                targetResourceGroup=inspect_result["targetResourceGroup"],
+                                targetImageName=inspect_result["image_name"],
+                                oldRunImage=run_images_tag,
+                                newRunImage=None,
+                                id=None,
+                                reason=failed_reason))
+                    else:
+                        # Not based on image from mcr.microsoft.com/oryx/builder
+                        results.append(dict(
+                            targetContainerAppName=inspect_result["targetContainerAppName"],
+                            targetContainerAppEnvironmentName=inspect_result["targetContainerAppEnvironmentName"],
+                            targetResourceGroup=inspect_result["targetResourceGroup"],
+                            oldRunImage=inspect_result["remote_info"]["run_images"],
+                            newRunImage=None,
+                            id=None,
+                            reason=mcr_check_reason))
+    # Make sure that we clear the first line before showing the new output
+    print("\r                                   \r", end="", flush=True)
+    if show_all is False:
+        results = [result for result in results if result["id"] is not None]
+    if not results:
+        logger.warning("No container apps available to patch at this time. Use --show-all to show the container apps that cannot be patched.")
+        return
+    return results
+
+
+def patch_get_image_inspection(pack_exec_path, img, info_list):
+    if (img["imageName"].find("run-dotnet") != -1) and (img["imageName"].find("cbl-mariner") != -1):
+        inspect_result = {
+            "remote_info":
+            {
+                "run_images":
+                [{
+                    "name": "mcr.microsoft.com/oryx/builder:" + img["imageName"].split(":")[-1]
+                }]
+            },
+            "image_name": img["imageName"],
+            "targetContainerName": img["targetContainerName"],
+            "targetContainerAppName": img["targetContainerAppName"],
+            "targetContainerAppEnvironmentName": img["targetContainerAppEnvironmentName"],
+            "targetResourceGroup": img["targetResourceGroup"]
+        }
+    else:
+        with subprocess.Popen(pack_exec_path + " inspect-image " + img["imageName"] + " --output json", shell=True, stderr=subprocess.PIPE, stdout=subprocess.PIPE) as img_info:
+            img_info_out, img_info_err = img_info.communicate()
+            if img_info_err.find(b"status code 401 Unauthorized") != -1 or img_info_err.find(b"unable to find image") != -1:
```

Comment:
is it possible to get a property like statusCode for img_info_err

Reply:
Yea the error handling here is not ideal to me as well. However, buildpack CLI doesn't have any return properties here. The only way to detect/handle the error is by looking through the stderr output.